include/z64actor.h

```diff
@@ -232,53 +252,88 @@ typedef struct {
+    /* 0x05F9 */ char       unk_5F9[0x06B];
     /* 0x0664 */ Actor*     unk_664;
     /* 0x0668 */ char       unk_668[0x004];
     /* 0x066C */ s32        unk_66C;
-    /* 0x0670 */ char       unk_670[0x00C];
+    /* 0x0670 */ u32        unk_670; // (sword?) effect related
+    /* 0x0674 */ ActorFunc  actionFunc;
+    /* 0x0678 */ u32        ageProperties;
     /* 0x067C */ u32        stateFlags1;
     /* 0x0680 */ u32        stateFlags2;
-    /* 0x0684 */ char       unk_684[0x008];
-    /* 0x068C */ Actor*     unk_68C;
-    /* 0x0690 */ char       unk_690[0x002];
+    /* 0x0684 */ s32        unk_684;
+    /* 0x0688 */ char       unk_688[0x004];
+    /* 0x068C */ Actor*     navi;
+    /* 0x0690 */ u16        naviMessageId;
     /* 0x0692 */ u8         unk_692;
     /* 0x0693 */ s8         exchangeItemId;
-    /* 0x0694 */ Actor*     unk_694;
-    /* 0x0698 */ f32        unk_698;
+    /* 0x0694 */ Actor*     naviTargetActor;
+    /* 0x0698 */ f32        targetActorDistance;
     /* 0x069C */ char       unk_69C[0x008];
     /* 0x06A4 */ f32        unk_6A4;
     /* 0x06A8 */ Actor*     unk_6A8;
     /* 0x06AC */ char       unk_6AC[0x001];
     /* 0x06AD */ u8         unk_6AD;
-    /* 0x06AE */ char       unk_6AE[0x1A];
+    /* 0x06AE */ char       unk_6AE[0x2];
+    /* 0x06B0 */ s16        unk_6B0;
+    /* 0x06B2 */ char       unk_6B4[0x4];
+    /* 0x06B6 */ Vec3s      unk_6B6;
+    /* 0x06BC */ s16        unk_6BC;
+    /* 0x06BE */ s16        unk_6BE;
+    /* 0x06C0 */ s16        unk_6C0;
+    /* 0x06C2 */ s16        unk_6C2;
+    /* 0x06C4 */ f32        unk_6C4;
     /* 0x06C8 */ SkelAnime  skelAnime2;
     /* 0x070C */ char       unk_70C[0x128];
     /* 0x0834 */ s16        unk_834;
     /* 0x0836 */ char       unk_836[0x002];
-    /* 0x0838 */ f32        unk_838;
-    /* 0x083C */ s16        unk_83C;
-    /* 0x083E */ char       unk_83E[0x004];
+    /* 0x0838 */ f32        linearVelocity;
+    /* 0x083C */ s16        currentYaw;
+    /* 0x083E */ s16        targetYaw;
+    /* 0x0840 */ u16        unk_840;
     /* 0x0842 */ s8         swordAnimation;
     /* 0x0843 */ s8         swordState;
     /* 0x0844 */ u8         unk_844;
     /* 0x0845 */ u8         unk_845;
     /* 0x0846 */ u8         unk_846;
     /* 0x0847 */ char       unk_847[0x004];
-    /* 0x084B */ s8         unk_84B[UNK_SIZE];
-    /* 0x084C */ char       unk_84C[0x003];
+    /* 0x084B */ s8         unk_84B[4];
     /* 0x084F */ s8         unk_84F;
-    /* 0x0850 */ char       unk_850[0x050];
+    /* 0x0850 */ char       unk_850[0x00C];
+    /* 0x085C */ f32        stickLength;
+    /* 0x0860 */ s16        stickFlameTimer;
+    /* 0x0862 */ s8         overheadItemId;
+    /* 0x0863 */ char       unk_863[0x021];
+    /* 0x0884 */ f32        ledgeDistance;
+    /* 0x0888 */ f32        wallDistance;
+    /* 0x088C */ char       unk_88C[0x008];
+    /* 0x0894 */ s16        dropY;
+    /* 0x0896 */ s16        dropDistance;
+    /* 0x0898 */ char       unk_898[0x008];
```

Comment:
maybe some comments to explain what these are? arent very self explanatory by the names, though maybe it will make more sense in context of the code

Reply:
are ledge and wall distance updated all times? maybe something like distFromLedge and distFromWall are more self explanatory

dropY is y position you left the ground from? maybe dropStartHeight (or Y)

only use these if they make sense, just feel like theres prob a name that tells you the whole story without even needing comments